core/src/main/java/nl/nn/adapterframework/http/HttpListenerServlet.java

```diff
@@ -55,42 +56,44 @@ public void init() throws ServletException {
+	public void invoke(Message message, HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
 		ISecurityHandler securityHandler = new HttpSecurityHandler(request);
 		try (PipeLineSession messageContext= new PipeLineSession()) {
 			messageContext.setSecurityHandler(securityHandler);
 			messageContext.put("httpListenerServletRequest", request);
 			messageContext.put("httpListenerServletResponse", response);
 			String service=request.getParameter(SERVICE_ID_PARAM);
-			Enumeration<String> paramnames=request.getParameterNames();
-			while (paramnames.hasMoreElements()) {
-				String paramname = paramnames.nextElement();
-				String paramvalue = request.getParameter(paramname);
-				if (log.isDebugEnabled()) {
-					log.debug("HttpListenerServlet setting parameter ["+paramname+"] to ["+paramvalue+"]");
-				}
-				messageContext.put(paramname, paramvalue);
+			Enumeration<String> paramNames=request.getParameterNames();
+			while (paramNames.hasMoreElements()) {
+				String paramName = paramNames.nextElement();
+				String paramValue = request.getParameter(paramName);
+				log.debug("HttpListenerServlet setting parameter [{}] to [{}]", paramName, paramValue);
+				messageContext.put(paramName, paramValue);
 			}
 			try {
 				log.debug("HttpListenerServlet calling service ["+service+"]");
-				String result=sd.dispatchRequest(service, message, messageContext);
-				response.getWriter().print(result);
-			} catch (ListenerException e) {
-				log.warn("HttpListenerServlet caught exception, will rethrow as ServletException",e);
-				response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR,e.getMessage());
+				Message result = sd.dispatchRequest(service, message, messageContext);
+				if (result.isBinary()) {
+					StreamUtil.copyStream(result.asInputStream(), response.getOutputStream(), 16384);
+				} else {
+					StreamUtil.copyReaderToWriter(result.asReader(), response.getWriter(), 16384);
+				}
+			} catch (ListenerException | IOException e) {
+				log.warn("HttpListenerServlet caught exception, will rethrow as ServletException", e);
+				response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, e.getMessage());
 			}
 		}
 	}
 
 	@Override
 	public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
-		String message=request.getParameter(MESSAGE_PARAM);
-		invoke(message,request,response);
+		String message = request.getParameter(MESSAGE_PARAM);
+		invoke(Message.asMessage(message), request,response);
 	}
 
 	@Override
 	public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
-		String message= StreamUtil.streamToString(request.getInputStream(),"\n",false);
+		Message message = Message.asMessage(request.getInputStream());
 		invoke(message,request,response);
```

Comment:
Eigenlijk moeten we dan hier ook het contentype uitlezen. Zie `MessageUtils.getContext(...)`